### PR TITLE
feat(DCS): add resource dcs account

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:
add resource dcs account

**Special notes for your reviewer**:

**Release note**:

```release-note
add resource dcs account
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dcs/ TESTARGS='-run TestAccDcsAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs/ -v -run TestAccDcsAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsAccount_basic
=== PAUSE TestAccDcsAccount_basic
=== CONT  TestAccDcsAccount_basic
--- PASS: TestAccDcsAccount_basic (187.93s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       187.975s

```

### CODE REVIEW

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the password of the account.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the description of the account.`,
+			},
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: can we update the description to empty? if so, should remove the computed.

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+		Delay:        2 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+
+	account, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the account (%s) to complete: %s", accountName, err)
+	}
+
+	d.SetId(utils.PathSearch("account_id", account, "").(string))
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: we should return an error is account can't be found

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go (hunk 1 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: yes, removed Computed.

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go (hunk 2 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: done

Comment (Jason-Zhang9309) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: I mean update the code like this
```suggestion
        accountID := utils.PathSearch("account_id", account, nil)
        if accountID == nil {
		return diag.Errorf("the account (%s) is not found: %s", accountName, err)
	}
	d.SetId(utils.PathSearch(accountID.(string))
```

Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: done

File: docs/resources/dcs_account.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of the DCS instance.
+  Changing this creates a new resource.
+
+* `account_name` - (Required, String, ForceNew) Specifies the name of the account.
+  Changing this creates a new resource.
+
+* `account_password` - (Required, String) Specifies the password of the account.
+
+* `account_role` - (Required, String) Specifies the role of the account. Value options: **read**, **write**.
```
Comment (houpeng80) [inline docs/resources/dcs_account.md]: should add description to the value options, write indicates the account has read and write privilege

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+		Target:       []string{"AVAILABLE"},
+		Refresh:      accountStatusRefreshFunc(instanceId, accountName, createAccountClient),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        2 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+
+	account, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the account (%s) to complete: %s", accountName, err)
```
Comment (houpeng80) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]:
```suggestion
		return diag.Errorf("error waiting for creating account (%s) to complete: %s", accountName, err)
```

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+
+	updateAccountProduct := "dcs"
+	updateAccountClient, err := cfg.NewServiceClient(updateAccountProduct, region)
+
+	if err != nil {
+		return diag.Errorf("error creating DCS client: %s", err)
+	}
+
+	if d.HasChange("account_password") {
+		resetPassword(ctx, d, updateAccountClient)
```
Comment (houpeng80) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: the error message of the return should not be ingored, and fix others

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go [hunk 6]
```diff
@@ -0,0 +1,521 @@
...
+		if err != nil {
+			return nil, "QUERY ERROR", err
+		}
+		getAccountBody, err := utils.FlattenResponse(getAccountResp)
+		if err != nil {
+			return nil, "PARSE ERROR", err
+		}
+
+		account := utils.PathSearch(fmt.Sprintf("accounts|[?instance_id =='%s' && account_name == '%s']|[0]", instanceId, accountName),
+			getAccountBody, nil)
```
Comment (houpeng80) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: the request path has contains instance_id, is it needed here?

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go [hunk 7]
```diff
@@ -0,0 +1,521 @@
...
+			"account_role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the role of the account.`,
+			},
+			"account_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the password of the account.`,
+			},
```
Comment (houpeng80) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: sensitive should be added

File: docs/resources/dcs_account.md (hunk 3 above)
Comment (346010762) [reply docs/resources/dcs_account.md]: done

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go (hunk 4 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: done

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go (hunk 5 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: done

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go (hunk 6 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: deleted instance_id

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go (hunk 7 above)
Comment (346010762) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go]: added

**Pull request merged**